Improve permission error messages in pdb and asyncio.tools

ivonastojanovic opened the pull request:
This PR enhances the user experience when a `PermissionError` occurs while attempting to attach to a process using `pdb -p` or tools like `asyncio ps` and `asyncio pstree`. It adds platform-specific guidance for Linux, macOS, and Windows to help users understand why the operation failed and how to resolve it (e.g., using `sudo`, adjusting system settings, or enabling debugging privileges).

Currently, the same logic is duplicated in both `pdb` and `asyncio.tools`. If anyone has recommendations on how to avoid this duplication, I’d appreciate input. I couldn't find a clear shared location in the standard library where this kind of utility logic would belong.

kumaraditya303 commented on Lib/asyncio/tools.py:
```diff
@@ -184,6 +185,83 @@ def _print_cycle_exception(exception: CycleFoundException):
+    This function is called when a PermissionError is encountered while trying
+    to attach to a process.
+    """
+    system = platform.system()
```
sys is already imported, so you can use `sys.platform`

ivonastojanovic committed:
Add detailed permission requirements for remote debugging

Add OS-specific permission requirements under the remote debugging
attachment protocol section. Include guidance for Linux, macOS, and
Windows to help users resolve common permission-related issues when
attaching to processes.

ivonastojanovic committed:
Show troubleshooting URL for remote debugging permission errors in pdb

Add help text and display a URL in permission-related errors when
using the -p option in pdb to assist users in resolving common
issues during remote debugging.

ivonastojanovic committed:
Show permission troubleshooting URL in asyncio.tools

Print a URL with steps to resolve permission errors when inspecting
processes using asyncio.tools.
